js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -316,55 +352,31 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
                               (async () => {
                                   transaction.feePayer ||= this.publicKey ?? undefined;
                                   if (transaction.recentBlockhash == null) {
-                                      let targetCommitment: Finality;
-                                      switch (connection.commitment) {
-                                          case 'confirmed':
-                                          case 'finalized':
-                                          case 'processed':
-                                              targetCommitment = connection.commitment;
-                                              break;
-                                          default:
-                                              targetCommitment = 'finalized';
-                                      }
-                                      let targetPreflightCommitment: Finality;
-                                      switch (options?.preflightCommitment) {
-                                          case 'confirmed':
-                                          case 'finalized':
-                                          case 'processed':
-                                              targetPreflightCommitment = options.preflightCommitment;
-                                              break;
-                                          case undefined:
-                                              targetPreflightCommitment = targetCommitment;
-                                          default:
-                                              targetPreflightCommitment = 'finalized';
-                                      }
-                                      const preflightCommitmentScore =
-                                          targetPreflightCommitment === 'finalized'
-                                              ? 2
-                                              : targetPreflightCommitment === 'confirmed'
-                                              ? 1
-                                              : 0;
-                                      const targetCommitmentScore =
-                                          targetCommitment === 'finalized'
-                                              ? 2
-                                              : targetCommitment === 'confirmed'
-                                              ? 1
-                                              : 0;
                                       const { blockhash } = await connection.getLatestBlockhash({
-                                          commitment:
-                                              preflightCommitmentScore < targetCommitmentScore
-                                                  ? targetPreflightCommitment
-                                                  : targetCommitment,
+                                          commitment: getTargetCommitment(),
```

Comment:
Probably should pass minContextSlot to getLatestBlockhash also. But if there is no minContextSlot to pass, we should arguably call getLatestBlockhashAndContext instead and start using the context slot we get back as the min.

Reply:
Sweet. Will do this in a follow up PR.